swarm/storage/mru/resource.go

```diff
@@ -121,139 +138,93 @@ type resource struct {
 	frequency  uint64
 	version    uint32
 	data       []byte
+	publicKey  [publicKeyLength]byte
```

Comment:
Why not store the address of the public key in the metachunk instead of the public key? It is smaller and supposedly safer.

Reply:
Sounds like an idea. I'll give it a stab!